### PR TITLE
Update infrastructure Intel PSW/DCAP version

### PR DESCRIPTION
**Pull request by CyanDevs**
This updates our Jenkins CI agent infrastructure with new Intel PSW/DCAP versions, and includes various updates / bugfixes to support it.

# Updates

* Linux (Ubuntu) now uses Intel PSW 2.15.101.1 and DCAP 1.12.101.1
* Windows Server 2019 now uses Intel PSW 2.14.101.1 and DCAP 1.12.101.1
* Azure VM image for Ubuntu 18.04 is now on kernel 5.4 with SGX driver 1.33.2
* Azure VM image for Ubuntu 20.04 is now on kernel 5.11 with SGX driver 1.41
* VM image builds now use OpenEnclaveJenkinsLibrary
* Bors runs now support use of OpenEnclaveJenkinsLibrary

# Bugfixes
* Fixes `generate_devkit` script. Now includes instructions/dependency information. Removes broken build for grapeboard platform on OPTEE_OS builds
* Fixes devkits reference to use a rebuilt copy of the previous devkit for building new Docker images
* Fixes broken VM image builds by readding user id / group id
* Fixes sgx_prv permissions to Jenkins build user to support driver 1.41 (Though this also needs to be reimplemented in Jenkins during agent provisioning)
* Fixes bug where oetools latest tag was not pushed to internal repo

### CODE REVIEW

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -1,11 +1,7 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
-OECI_LIB_VERSION = env.OECI_LIB_VERSION ?: "master"
-oe = library("OpenEnclaveCommon@${OECI_LIB_VERSION}").jenkins.common.Openenclave.new()
-
-GLOBAL_TIMEOUT_MINUTES = 120
-CTEST_TIMEOUT_SECONDS = 480
+library "OpenEnclaveJenkinsLibrary@cyan/intel-update"
```
Comment (CyanDevs) [inline .jenkins/Jenkinsfile:4]: To be changed back to master before merging

File: .jenkins/library/vars/tests.groovy [hunk 2]
```diff
@@ -59,6 +59,9 @@ def ACCTest(String label, String compiler, String build_type, List extra_cmake_a
                         done
                         """
                 }
+                sh """
+                sudo apt list --installed | grep sgx
```
Comment (CyanDevs) [inline .jenkins/library/vars/tests.groovy:63]: Nit: spacing

File: scripts/ansible/roles/linux/jenkins/tasks/packer.yml [hunk 3]
```diff
@@ -36,6 +36,19 @@
...
+    key: sgx_prv
+
+- name: Add user to sgx_prv group
+  user:
+    name: "{{ jenkins_admin_name }}"
+    groups: sgx_prv
+    append: yes
+  when: "'sgx_prv' in ansible_facts.getent_group"
+
+
```
Comment (CyanDevs) [inline scripts/ansible/roles/linux/jenkins/tasks/packer.yml:51]: Nit: spacing

File: scripts/ansible/roles/linux/jenkins/tasks/packer.yml [hunk 4]
```diff
@@ -36,6 +36,19 @@
     groups: docker
     append: yes
 
+- name: Determine if sgx_prv group exists
+  getent:
+    database: group
+    key: sgx_prv
+
+- name: Add user to sgx_prv group
```
Comment (CyanDevs) [inline scripts/ansible/roles/linux/jenkins/tasks/packer.yml:44]: Should add comment that this step is often moot and needs to also be implemented during Jenkins provisioning

File: scripts/generate-devkits [hunk 5]
```diff
@@ -38,9 +68,9 @@ fi
 if [ ! -d optee/vexpress-qemu_armv8a ]; then
     mkdir optee/vexpress-qemu_armv8a
 fi
-if [ ! -d optee/ls-ls1012grapeboard ]; then
-    mkdir optee/ls-ls1012grapeboard
-fi
+# if [ ! -d optee/ls-ls1012grapeboard ]; then
```
Comment (CyanDevs) [inline scripts/generate-devkits:71]: Delete if determined to not be needed

**Review by radhikaj — APPROVED** [review]

**Comment by CyanDevs** [comment]
bors try

**Comment by CyanDevs** [comment]
bors try

**Review by achamayou — APPROVED** [review]

**Review by rs-- — APPROVED** [review]

**Pull request merged**